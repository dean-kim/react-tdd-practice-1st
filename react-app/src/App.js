import React from 'react';
import ReactDOM from 'react-dom';

class App extends React.Component {
    constructor() {
        super();
        this.state = {a: ''}
    }
    update() {
        this.setState({
            a: this.a.refs.input.value,
            b: this.b.value
        })
    }
    render() {

        return (
            <div>
                <Input
                    ref={ component => this.a = component}
                    type="text"
                    update={this.update.bind(this)}/>
                {this.state.a}
                <hr />
                <input
                    ref={ node => this.b = node}
                    type="text"
                    onChange={this.update.bind(this)}/>
                {this.state.b}
            </div>
        )
    }
}


class Input extends React.Component {
    render() {
        return (
            <div>
                <input ref="input" type="text" onChange={this.props.update} />
            </div>
        )
    }
}



export default App