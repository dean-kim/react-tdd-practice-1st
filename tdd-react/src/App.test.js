import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<App />', () => {
    it('should contain ul element className "tyler"', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('ul').hasClass('tyler')).toBe(true)
    })
});
