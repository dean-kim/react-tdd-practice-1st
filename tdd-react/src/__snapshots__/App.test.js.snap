// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <p
      className="App-intro"
    >
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
